### ISSUE TITLE
Disconnected by libera with "Excess Flood" if joining too many channels

### ISSUE THREAD

**Issue by popey** [squidowl/halloy#256]
If I connect to Libera using Halloy and configure more than ~10 channels, Libera instantly disconnects me. I can connect successfully if I reduce the number of channels I auto-join below ~10.

```
20:06:18.234:DEBUG -- loaded history for channel #halloy on liberachat: 45 messages
20:06:18.234:DEBUG -- loaded history for user casperstorm on liberachat: 19 messages
20:06:18.234:DEBUG -- loaded history for server on liberachat: 692 messages
20:06:18.549:INFO -- [liberachat] connected
20:06:24.470:INFO -- [liberachat] capabilities acknowledged: away-notify server-time sasl
20:06:24.796:WARN -- [liberachat] disconnected: Closing Link: myhost.virginm.net (Excess Flood)
20:06:30.207:DEBUG -- flushed history for channel #halloy on liberachat
20:06:30.207:DEBUG -- flushed history for user casperstorm on liberachat
20:06:30.207:INFO -- dashboard saved
20:06:30.212:DEBUG -- flushed history for server on liberachat
20:06:33.405:DEBUG -- closed history for channel #halloy on liberachat
20:06:33.405:DEBUG -- closed history for user casperstorm on liberachat
20:06:33.405:DEBUG -- closed history for server on liberachat
```

```
alan@nuc:~$ halloy --version
halloy 2024.3 (bc4416a)
```

## Steps to reproduce

1. Configure halloy to connect directly to libera
2. Add a bunch of unused channel names to the configuration
3. Launch halloy
4. Observe the problem
5. Close halloy
6. Reduce the number of channels in the libera section
7. Goto 3

```
servers:
  # Configuration for Libera server
  liberachat:
    # Nickname to be used on the server
    nickname: (myusername)
    realname: "My Name"

    # Server address
    server: irc.libera.chat

    # Server port number
    port: 6697

    # Whether to use TLS
    use_tls: true

    # Channels to join upon connecting to the server
    channels:
      - "#halloy"
      - '##popeytest-01'
      - '##popeytest-02'
      - '##popeytest-03'
      - '##popeytest-04'
      - '##popeytest-05'
      - '##popeytest-06'
      - '##popeytest-07'
      - '##popeytest-08'
      - '##popeytest-09'
      - '##popeytest-10'
      - '##popeytest-11'
      - '##popeytest-12'
      - '##popeytest-13'
      - '##popeytest-14'
      - '##popeytest-15'
      - '##popeytest-16'
      - '##popeytest-17'
      - '##popeytest-18'
      - '##popeytest-19'
      - '##popeytest-20'
    on_connect:
      - "/msg Nickserv identify (mypassword) (mynickname)"
```

**Comment by casperstorm** [comment]
@popey thanks, i can reproduce this.
We should have a code path to handle something similar, so will start investigating a fix for it.

**Comment by casperstorm** [comment]
Fixed in #269.
